### DISCUSSION TITLE
about externsion

### DISCUSSION THREAD

**Discussion by xubobbs** [duckdb/duckdb#15083] [Q&A]
hi,
I want to use some extension in my project .
but my backend db is configured by project ,and some db type not supported by duckdb .
I want to use  some "normal" extension , when duckdb call the scan ,I hope I can implment the query or other operate , and at that point I use the db driver to query and return the result to the scan...

how can I do ?
thanks

**Comment by Tishj** [comment]
You should take a look at the spatial source code, as that introduces a good amount of new types that are entirely handled by the spatial extension
The JSON extension might be a better starting point, as it only introduces the JSON type

**Reply by xubobbs** [reply]
got it ,thanks

**Comment by xubobbs** [comment]
I use sqlite externsion as demo and test .and now I has some question.
```
D  load 'build/debug/extension/sqlite_scanner/sqlite_scanner.duckdb_extension';
D attach '/home/xubo/migration.db' as migration_db (type sqlite);
select client_id from  migration_db.client_client_client_meta_v6 where client_client_client_meta_v6.update_time > 0 limit 1;
xubo debug Execute: BEGIN TRANSACTION
xubo debug PREPARE: SELECT type FROM sqlite_master WHERE lower(name)=lower('client_client_client_meta_v6');
xubo debug PREPARE: PRAGMA table_info('client_client_client_meta_v6')
xubo debug Execute: COMMIT
xubo debug Execute: BEGIN TRANSACTION
xubo debug PREPARE: SELECT type FROM sqlite_master WHERE lower(name)=lower('client_client_client_meta_v6');
xubo debug PREPARE: PRAGMA table_info('client_client_client_meta_v6')
xubo debug PREPARE: SELECT "update_time", "client_id" FROM "client_client_client_meta_v6"
xubo debug Execute: COMMIT
```
I add some debug code in sqlite externsion and print query sql.
my question is the following sql 
select client_id from  migration_db.client_client_client_meta_v6 where client_client_client_meta_v6.update_time > 0 limit 1;
why no "push down" conditon on sqlite remote db ? 
because the debug execute sql on sqlite is "SELECT "update_time", "client_id" FROM "client_client_client_meta_v6" with no any filter "update_time" ?

and the same question on join clause.

**Reply by Maxxen** [reply]
If I remember correctly we discovered that DuckDB is faster at executing filters than sqlite is, so it is better to not push it down into sqlite.

**Reply by xubobbs** [reply]
but I want to study how to push down ,because maybe my database is not sqlite and I want to do some push down filter...
